actors/evm/src/interpreter/instructions/ext.rs

```diff
@@ -12,18 +12,20 @@ use {
     fil_actors_runtime::runtime::Runtime,
 };
 
-#[inline]
 pub fn extcodesize(
     _state: &mut ExecutionState,
     system: &System<impl Runtime>,
     addr: U256,
 ) -> Result<U256, StatusCode> {
-    // TODO we're fetching the entire block here just to get its size. We should instead use
+    // TODO (M2.2) we're fetching the entire block here just to get its size. We should instead use
     //  the ipld::block_stat syscall, but the Runtime nor the Blockstore expose it.
     //  Tracked in https://github.com/filecoin-project/ref-fvm/issues/867
-    let len = get_evm_bytecode_cid(system.rt, addr)
-        .and_then(|cid| get_evm_bytecode(system.rt, &cid))
-        .map(|bytecode| bytecode.len())?;
+    let len = match get_cid_type(system.rt, addr) {
+        CodeCid::EVM(addr) => get_evm_bytecode(system.rt, &addr).map(|bytecode| bytecode.len())?,
+        CodeCid::Native(_) => 1,
```

Comment:
why 1?

Reply:
ah, 0xfe below.

Reply:
yes, but has more to do with playing nicely with solidity's behavior of `isContract`